src/context/notificationStateSlice.ts

```diff
@@ -84,23 +84,39 @@ export const {
 export const errorToNotificationMiddleware: Middleware<{}, RootState> = () => (
   dispatch: AppDispatch,
 ) => (action: AnyAction) => {
-  const dispatchResult = dispatch(action);
-
-  // eslint-disable-next-line default-case
-  switch (action.type) {
-    case 'mapState/loadLayerData/rejected':
-    case 'serverState/loadAvailableDates/rejected':
-    case 'tableState/loadTable/rejected':
-      dispatch(
-        addNotification({
-          type: 'error',
-          message:
-            typeof action.error === 'string'
-              ? action.error
-              : action.error.message,
-        }),
-      );
+  let dispatchResult;
+  try {
+    // catch sync errors
+    // eslint-disable-next-line fp/no-mutation
+    dispatchResult = dispatch(action);
+  } catch (err) {
+    console.error(err);
+    dispatch(
+      addNotification({
+        type: 'error',
+        message: err.message || err,
+      }),
+    );
+    return dispatchResult;
   }
+  const errorActions = [
+    // async thunks that have errors included with them
+    'tableState/loadTable/rejected',
+    'serverState/loadAvailableDates/rejected',
+    'mapState/loadLayerData/rejected',
+  ];
+  if (errorActions.includes(action.type)) {
```

Comment:
Maybe just do a regex on this so we don't have to remember to enumerate these everytime we add an async thunk?